common/sku_customization.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -ex
+
+# setup systemd service
+cat <<EOF >/usr/sbin/setup_topo.sh
+#!/bin/bash
+
+metadata_endpoint="http://169.254.169.254/metadata/instance?api-version=2019-06-04"
+vmSize=\$(curl -H Metadata:true \$metadata_endpoint \
+    | grep -Eo '"vmSize"[^,]*' \
+    | awk  -F'"' '{print \$4}')
+
+nccl_conf="/etc/nccl.conf"
+case \$vmSize in
+    Standard_NC*ads_A100_v4)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ncv4/topo.xml" >> \$nccl_conf;;
+    
+    Standard_ND96*v4)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ndv4/topo.xml" >> \$nccl_conf;;
+        
+    Standard_ND40rs_v2)
+        echo "NCCL_TOPO_FILE=/opt/microsoft/ndv2/topo.xml" >> \$nccl_conf;;
+
+    *) echo "No Topo file for \$vmSize"; exit 1;;
+esac
+EOF
+chmod 755 /usr/sbin/setup_topo.sh
+
+cat <<EOF >/etc/systemd/system/setup_topo.service
+[Unit]
+Description=Setup appropriate topo file for the SKU
+After=network.target
+
+[Service]
+Type=oneshot
+ExecStart=/usr/sbin/setup_topo.sh
+RemainAfterExit=true
+StandardOutput=journal
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+systemctl enable setup_topo
+systemctl start setup_topo
```

Comment:
Can we also check the service status to ensure it's started successfully?

Reply:
Added `systemctl is-active --quiet setup_topo` for status check
It will exit with status zero if service is active, non-zero otherwise

Reply:
Good idea. Not really related to this PR, but we might want to add this to some other services as well at some point. 

https://github.com/Azure/azhpc-images/blob/master/common/install_azure_persistent_rdma_naming.sh#L69